core/src/main/java/org/mule/runtime/core/internal/context/notification/DefaultFlowCallStack.java

```diff
@@ -28,19 +24,14 @@
 
   private static final long serialVersionUID = -8683711977929802819L;
 
-  private static final Logger LOGGER = getLogger(DefaultFlowCallStack.class);
-
-  private Stack<FlowStackElement> innerStack = new Stack<>();
+  private final Stack<FlowStackElement> innerStack = new Stack<>();
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace the synchronized class "Stack" by an unsynchronized one such as "Deque". ([squid:S1149](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1149))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7712&issues=AWkR5Jd8otIXnmsUahsx&open=AWkR5Jd8otIXnmsUahsx)

[ISSUE_KEY]: AWkR5Jd8otIXnmsUahsx